metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -796,6 +851,39 @@ def __init__(
         # when emitting custom SQL data sources.
         self.custom_sql_ids_being_used: List[str] = []
 
+        self.report_user_role()
+
+    def report_user_role(self):
+        title: str = "Insufficient Permissions"
+        message: str = 'The user must have the "SiteAdministratorExplorer" role to perform metadata ingestion.'
+        try:
+            # TableauSiteSource instance is per site, so each time we need to find-out user detail
+            # the site-role might be different on another site
+            logged_in_user: LoggedInUser = LoggedInUser(server=self.server)
+
+            if not logged_in_user.is_site_administrator_explorer():
+                self.report.warning(
+                    title=title,
+                    message=message,
+                    context=f"user-name={logged_in_user.user_name()}, role={logged_in_user.site_role()}, site_id={logged_in_user.site_id()}",
+                )
+
+            self.report.user_site_info.append(
```

Comment:
can we not just do this as the first step in the main TableauSource?

The main purpose of this check is to ensure that we know the permissions we have before we've made any other API calls. We make an api call to list servers in the TableauSource, so this should go before that

Reply:
for login to work site_id is required. I added the call to report_user_role in initial `def _authenticate(self, site_content_url: str) -> None:` method as well.